File: solutions/dart/pangram/1/lib/pangram.dart
class Pangram {
  bool isPangram(String str) {
    List<bool> appears = List.filled(26, false);

    for (var c in str.toLowerCase().split('')) {
      var index = c.codeUnitAt(0) - 'a'.codeUnitAt(0);
      if (index < 0 || index > 26) continue;

      appears[index] = true;
    }

    return appears.every((c) => c == true);
  }
}

File: solutions/javascript/pangram/1/pangram.js
//
// This is only a SKELETON file for the 'Pangram' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isPangram = (sentence) => {
  const alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
  const sentenceLower = sentence.toLowerCase();
  const sentenceArray = sentenceLower.split('');

  return alphabet.every((letter) => sentenceArray.includes(letter));
};
